dump/main.c

```diff
@@ -140,9 +140,83 @@ static int dump_full(const char* spec, bool used_sectors)
 	return 0;
 }
 
+static int dump_file(const char* spec, const char* path)
+{
+	struct exfat ef;
+	struct exfat_node* node;
+	cluster_t cluster, next_cluster;
+	cluster_t start_cluster = EXFAT_CLUSTER_FREE;
+	off_t remainder;
+	off_t extent_size;
+	off_t lsize;
+	int rc = 1;
+
+	if (exfat_mount(&ef, spec, "ro") != 0)
+		return 1;
+
+	if (exfat_lookup(&ef, &node, path) != 0)
+	{
+		exfat_unmount(&ef);
+		exfat_error("Failed to lookup path: %s", path);
+		return 1;
+	}
+
+	cluster = exfat_advance_cluster(&ef, node, 0);
+	if (CLUSTER_INVALID(cluster))
```

Comment:
This code will fail on empty files. What output do you expect for a zero-sized file?

BTW you can get the first cluster from `node->start_cluster`.